### ISSUE TITLE
Doom 3 BFG VR on Oculus Quest ?

### ISSUE THREAD

**Issue by motorsep** [KozGit/DOOM-3-BFG-VR#299]
Is there even remote possibility that this port can run on Oculus Quest ?

**Comment by Midnightmaster** [comment]
I think there are three problems:
1 KozGit is not unemployed (I think so :-D ) and that would be a very big project. Only with support from Kickstarter possible.
2. the performance of Doom 3 BFG VR is not very good. Only with my 1080ti and a core i7-8700k it runs really smooth. The GPU/CPU of the Quest is extremely slow compared to a PC. All ports of the Quest (Robo Recall) look extremely trimmed in the graphics.
3. to play the mod you need the PC retail version of Doom 3 BFG (Steam!?!). On Quest there is no version you can buy. Offering the complete ( new compiled ) files on Quest would probably be a punishable offence if I see that correctly. Maybe you can fix it differently if you already have the Steam version and only have to copy certain files.

Since I will also get the Quest for release, I would of course be very pleased with a Quest version. In terms of playing time it would probably be the longest game on the quest until Skyrim is released for the quest.

**Comment by motorsep** [comment]
@Midnightmaster about that.. 

#2 isn't quite accurate as there was a port of Doom 3 BFG running on Galaxy S8 with lights and shadows at 60 fps. So I am sure it's possible to have it running on Quest with required performance.

#3 is irrelevant - buy DOom 3 BFG on Steam, copy data files to your Quest. Executables (engine mod) have nothing to do with where you get the data. Just how it works with Quake 1 for Oculus Go.

While #1 is probably the most critical factor here, if Doom 3 BFG VR uses Oculus SDK, adding support for Quest shouldn't be as complicated as it might appear to be. Most of the work would probably go into the ES3.1 renderer. Hard to say really :(

So I am guess the answer is "no" :)

**Comment by Midnightmaster** [comment]
2. I think the S8 has more power than the Quest, but it can also mislead me. After all, the operating system is only designed for gaming and the S8 has to do hundreds of other tasks in the background. The ancient principle of the game console.
I took a look at it on Youtube: It seems to me the guy has set all settings to low (textures) to achieve an acceptable frame rate. So nobody wants to play the game on his quest in VR. But it does run smoothly. Unfortunately we need a frame rate of 144 (2 x 72 frames) on the quest without ASW. Doom 3 BFG also runs smoothly in most cases, unfortunately there are many places in the game where the frame rate drops massively in VR.

3. Yeah, you might be right about that, shouldn't be such a big deal.

**Comment by KozGit** [comment]
#3 If someone has the technical know how to get the mod onto the Quest in the first place, it shouldn't be difficult for them to copy the game files as well.  I do think sideloading **anything** onto the device may pose an issue for the general target audience of the Quest though.  

#2 I don't want to be negative, especially without any first hand knowledge of Quest performance, but I can easily imagine performance being a significant issue.  Would be fun to experiment :)

#1 Well, yeah, I'm definitely not unemployed :)   I accepted a new role about a year ago that's keeping me busy - and is the reason I haven't had much time to devote to this recently.  I have fun working on this when I have time, and have enjoyed the simplicity of keeping money out of it.  I don't know that a Kickstarter would help much ( at least it wouldn't help me much ) - it wouldn't add more hours to the day, and regardless of legality, I have difficulty imagining it would raise significant funds - certainly not enough to compete with a career.  I've had a few musings about there being enough interest in creating VR ports of some of these older games ( and there are several out there with source code ) to establish a decent Patreon, but I just don't think the segment of the VR community that would be interested in supporting this is large enough.  It's a fun daydream though :)

**Comment by motorsep** [comment]
Well, S8 and Quest have the same exact SoC in it - Qualcomm 835. The difference is that Quest has a way better thermals, overclocked SoC and barebones Android. So it's quite faster than S8.

Look, they got Robo Recall running on Quest and Crytek's Climb is coming there too.

Quest allows sideloading, just like Go. People play Quake 1 mod for Go, so I am sure anyone who does that can sideload Doom 3 BFG, if DOom 3 BFG would have a port.

So, it all comes down to availability.

**Comment by CarlKenner** [comment]
I am unemployed, but I probably won't port it to Quest (ie. Android).
I don't have a Quest, but I do have a GearVR for Galaxy S6. There is already an issue #271 thread on here about playing Doom 3 BFG: Fully Possessed on GearVR by streaming it from the PC, so I imagine the same thing will be done with Quest in the near future for people who don't have VR, or as good VR, on PC.

**Comment by motorsep** [comment]
The whole point is that Quest doesn't need a PC, at all, and it's a 6DoF HMD with inside-out tracking where you can walk in a park or stadium to move freely in VR. Also Quest cost less than a decent PC.

**Comment by CarlKenner** [comment]
My decent PC is broken right now, so I'm going to have to get it fixed/replaced rather than buying a Quest, sorry. Quest support isn't a priority, but it's not impossible in the future.

**Comment by sycspysycspy** [comment]
> 
> 
> I am unemployed, but I probably won't port it to Quest (ie. Android).
> I don't have a Quest, but I do have a GearVR for Galaxy S6. There is already an issue #271 thread on here about playing Doom 3 BFG: Fully Possessed on GearVR by streaming it from the PC, so I imagine the same thing will be done with Quest in the near future for people who don't have VR, or as good VR, on PC.

I am both happy and sad to hear that you are unemployed. Keep it together man.

**Comment by cercata** [comment]
Shit, a talented guy like CarlKenner doesn't have a computer to work ???? That's a loss for the VR Scene !!!!!

Regarding Quest, if it sells several millions Units, it will become an desired target for Sceners.
There is already a tool for making sideloading easier ...

Regarding Doom-3-BFG-VR, I think the priority should be release the 0.022 on PC

**Comment by CarlKenner** [comment]
I got my computer (mostly) working again, but I had to disable one of my hard disks in the BIOS.
0.022 will never be released because we have moved past that. The next released version will have a higher number and additional features that weren't intended for 0.022.

**Comment by jdawgzim** [comment]
I can't wait to try Doom 3 BFG VR on **Valve Index**.  I loved it on the Vive.  You guys have done amazing work and I wish we could pitch in to help ya'll with VR upgrades.  -But I understand the desire to keep **donations** out of the work.  It can ruin things.

**Comment by CactusVRStudios** [comment]
Can we expect an update anytime soon? This is by far the gold standard when it comes to vr mods.

**Comment by bigelod** [comment]
Out of pure curiosity I tried out the WASM Doom 3 / emscripten demo in the Oculus Quest browser, aside from input it actually worked (sound too but my video didn't capture it): https://www.youtube.com/watch?v=9zBxWfb1ziU&feature=youtu.be

**Comment by motorsep** [comment]
@bigelod Well, it's not VR and I can hardly call 14 fps a good performance. I guess streaming this mod to Quest using Virtual Desktop is an option, but the idea to have this on Quest natively is that you wouldn't need PC.

**Comment by bigelod** [comment]
That's running without optimization in a browser without even fixed foveated rendering

I didn't even tweak any Graphics settings

**Comment by motorsep** [comment]
It's 2D, running on the flat screen, as if you'd be playing pancake game in real life. FFR has no effect on it. FFR is for VR games.

**Comment by bigelod** [comment]
Literally my point.

**Comment by sycspysycspy** [comment]
There's some technology to allow quest streaming from a pc, so this kinda allows you to play many pc exclusive vr games including doom 3 vr on quest right?

**Comment by motorsep** [comment]
@sycspysycspy Yeah, it's called Virtual Desktop for Quest. Has to be sideloaded (version with WiFi streaming to HMD isn't on the official store). It allows playing all PCVR games on Quest, wirelessly. It has a bit of latency, but people swear it's almost like playing wired VR using Rift/Rift S. But again, you need VR-ready PC for this.
